### PR TITLE
feat: add support for restore token

### PR DESCRIPTION
**Pull request by thiyaguk09**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #restoreToken

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -547,6 +551,7 @@ class File extends ServiceObject<File, FileMetadata> {
   name: string;
 
   generation?: number;
+  restoreToken?: string;
```
Comment (ddelgrosso1) [inline src/file.ts:554]: Is there a reason to have this field as both a class level and metadata level field?

**Comment by ddelgrosso1** [comment]
My understanding is that the restore token would be returned on a call to delete. Is that going to be a separate PR or should it be included here?

**Comment by thiyaguk09** [comment]
> My understanding is that the restore token would be returned on a call to delete. Is that going to be a separate PR or should it be included here?

I'm currently focused on implementing support for restore tokens in the get, list, and restore operations.

File: src/file.ts (hunk 1 above)
Comment (thiyaguk09) [reply src/file.ts:554]: I've noticed that in some scenarios, a refresh token is associated with a specific generation ID. To ensure consistent behavior and avoid potential issues, I've implemented the refresh token handling in a way that aligns with the generation ID concept. This approach will help maintain data integrity and security.

**Review by ddelgrosso1 — APPROVED** [review]

**Pull request merged**